src/databricks/labs/remorph/reconcile/compare.py

```diff
@@ -178,3 +179,211 @@ def _get_mismatch_df(source: DataFrame, target: DataFrame, key_columns: list[str
+    ]
+    conditions = [
+        col(f"{source_alias}.{item[0]}").eqNullSafe(col(f"{target_alias}.{item[1]}")) for item in join_columns
+    ]
+    return reduce(lambda a, b: a & b, conditions)
+
+
+def _agg_conditions(
+    cols: list[tuple[str, str]] | None,
+    condition_type: str = "group_filter",
+    op_type: str = "and",
+):
+    """
+    Generate conditions for aggregated data comparison based on the condition type
+    and reduces it based on the operator (and, or)
+
+    e.g.,  cols = [(source_min_col1, target_min_col1)]
+              1. condition_type = "group_filter"
+                    source_group_by_col1 is not null and target_group_by_col1 is not null
+              2. condition_type = "select"
+                    source_min_col1 == target_min_col1
+              3. condition_type = "missing_in_src"
+                    source_min_col1 is null
+              4. condition_type = "missing_in_tgt"
+                      target_min_col1 is null
+
+    :param cols:  List of columns to compare
+    :param condition_type:  Type of condition to generate
+    :param op_type: and, or
+    :return:  Reduced column expressions
+    """
+    assert cols, "Columns must be specified for aggregation conditions"
+
+    conditions_list = [(col(f"{item[0]}").isNotNull() & col(f"{item[1]}").isNotNull()) for item in cols]
+
+    match condition_type:
+        case "select":
+            conditions_list = [col(f"{item[0]}") == col(f"{item[1]}") for item in cols]
+        case "missing_in_src":
+            conditions_list = [col(f"{item[0]}").isNull() for item in cols]
+        case "missing_in_tgt":
+            conditions_list = [col(f"{item[1]}").isNull() for item in cols]
+
+    return reduce(lambda a, b: a & b if op_type == "and" else a | b, conditions_list)
+
+
+def _generate_match_columns(select_cols: list[tuple[str, str]]):
+    """
+    Generate match columns for the given select columns
+    e.g.,  select_cols = [(source_min_col1, target_min_col1), (source_count_col3, target_count_col3)]
+            |--------------------------------------|---------------------|
+           |               match_min_col1                      |  match_count_col3 |
+           |--------------------------------------|--------------------|
+             source_min_col1 == target_min_col1 | source_count_col3 == target_count_col3
+           --------------------------------------|---------------------|
+
+    :param select_cols:
+    :return:
+    """
+    items = []
+    for item in select_cols:
+        match_col_name = item[0].replace("source_", "match_")
+        items.append((match_col_name, col(f"{item[0]}") == col(f"{item[1]}")))
+    return items
+
+
+def _get_mismatch_agg_data(
+    df: DataFrame,
+    select_cols: list[tuple[str, str]],
+    group_cols: list[tuple[str, str]] | None,
+) -> DataFrame:
+    # TODO:  Integrate with _get_mismatch_data function
```

Comment:
Can you add the link to the TODO

Reply:
https://github.com/databrickslabs/remorph/issues/745 has been created